editor/effects.js

```diff
@@ -20,9 +20,30 @@ import {
 	getCurrentPostType,
 	getBlocks,
 	getPostEdits,
+	isCleanNewPost,
 } from './selectors';
 
+let originalDocumentTitle;
+
+function populateDocumentTitle( title, isCleanNew = false ) {
+	if ( ! title.trim() ) {
+		title = isCleanNew ? __( 'New post' ) : __( '(Untitled)' );
+	}
+	if ( ! originalDocumentTitle ) {
```

Comment:
Potentially. However, that would require that we export a template from PHP that we can use in JS. This was a quicker way to achieve that goal. 

The Customizer implements this via:
* https://github.com/WordPress/wordpress-develop/blob/4.8.0/src/wp-includes/class-wp-customize-manager.php#L3630
* https://github.com/WordPress/wordpress-develop/blob/4.8.0/src/wp-admin/js/customize-controls.js#L3646-L3659

I don't feel it is entirely clear where we should be exporting data from PHP to JS generally.